crates/bevy_state/src/state/transitions.rs

```diff
@@ -57,26 +57,56 @@ pub struct StateTransitionEvent<S: States> {
 ///
 /// These system sets are run sequentially, in the order of the enum variants.
 #[derive(SystemSet, Clone, Debug, PartialEq, Eq, Hash)]
-pub enum StateTransitionSteps {
+pub(crate) enum StateTransitionSteps {
     /// Parentless states apply their [`NextState<S>`].
     RootTransitions,
     /// States with parents apply their computation and [`NextState<S>`].
     DependentTransitions,
-    /// Exit schedules are executed.
+    /// Exit schedules are executed in leaf-root order
```

Comment:
Tiny nit: I'd write this as `in leaf -> root order`, or `in leaf to root order`, or `from leaf to root`; otherwise it might read as a hyphen. Same for the other two doc comments.